File: lib/models/atendimento.dart
class Atendimento {
  final int id;
  final int idAluno;
  final int idFuncionario;

  final String motivoDaConsulta;
  final String anamnese;

  final double peso;
  final double altura;
  final double imc;

  final double perimetroCefalico;
  final double circunferenciaAbdominal;
  final double perimetroPanturrilha;

  final String exameFisico;
  final String diagnostico;

  final DateTime dataAtendimento;

  Atendimento({
    required this.id,
    required this.idAluno,
    required this.idFuncionario,
    required this.motivoDaConsulta,
    required this.anamnese,
    required this.peso,
    required this.altura,
    required this.imc,
    required this.perimetroCefalico,
    required this.circunferenciaAbdominal,
    required this.perimetroPanturrilha,
    required this.exameFisico,
    required this.diagnostico,
    required this.dataAtendimento,
  });

  factory Atendimento.fromJson(Map<String, dynamic> json) {
    return Atendimento(
      id: json['id'],
      idAluno: json['idAluno'],
      idFuncionario: json['idFuncionario'],
      motivoDaConsulta: json['motivoDaConsulta'],
      anamnese: json['anamnese'],
      peso: (json['peso'] as num).toDouble(),
      altura: (json['altura'] as num).toDouble(),
      imc: (json['imc'] as num).toDouble(),
      perimetroCefalico:
          (json['perimetroCefalico'] as num).toDouble(),
      circunferenciaAbdominal:
          (json['circunferenciaAbdominal'] as num).toDouble(),
      perimetroPanturrilha:
          (json['perimetroPanturrilha'] as num).toDouble(),
      exameFisico: json['exameFisico'],
      diagnostico: json['diagnostico'],
      dataAtendimento:
          DateTime.parse(json['dataAtendimento']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'idAluno': idAluno,
      'idFuncionario': idFuncionario,
      'motivoDaConsulta': motivoDaConsulta,
      'anamnese': anamnese,
      'peso': peso,
      'altura': altura,
      'imc': imc,
      'perimetroCefalico': perimetroCefalico,
      'circunferenciaAbdominal': circunferenciaAbdominal,
      'perimetroPanturrilha': perimetroPanturrilha,
      'exameFisico': exameFisico,
      'diagnostico': diagnostico,
      'dataAtendimento': dataAtendimento.toIso8601String(),
    };
  }
}

File: backend/Models/Atendimento.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace NeuroGest.API.Models
{
    public class Atendimento
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int IdAluno { get; set; }

        [Required]
        public int IdUsuario { get; set; }

        // Agendamento que originou este atendimento. Nulo apenas em
        // atendimentos antigos, anteriores a esta exigência — novos
        // atendimentos sempre exigem um agendamento prévio (validado no
        // controller, já que registros existentes impedem tornar a coluna
        // obrigatória no banco).
        public int? IdAgendamento { get; set; }

        [Required]
        public string MotivoDaConsulta { get; set; } = string.Empty;

        [Required]
        public string Anamnese { get; set; } = string.Empty;

        [Column(TypeName = "decimal(5,2)")]
        public decimal Peso { get; set; }

        [Column(TypeName = "decimal(4,2)")]
        public decimal Altura { get; set; }

        [Column(TypeName = "decimal(4,2)")]
        public decimal Imc { get; set; }

        [Column(TypeName = "decimal(4,2)")]
        public decimal PerimetroCefalico { get; set; }

        [Column(TypeName = "decimal(4,2)")]
        public decimal CircunferenciaAbdominal { get; set; }

        [Column(TypeName = "decimal(4,2)")]
        public decimal PerimetroPanturrilha { get; set; }

        [Required]
        public string ExameFisico { get; set; } = string.Empty;

        [Required]
        public string Diagnostico { get; set; } = string.Empty;

        [Required]
        public DateTime DataAtendimento { get; set; } = DateTime.UtcNow;

        // Propriedades de Navegação para o Entity Framework (Opcional, mas recomendado)
        [ForeignKey("IdAluno")]
        public virtual Aluno? Aluno { get; set; }

        [ForeignKey("IdUsuario")]
        public virtual Usuario? Usuario { get; set; }

        [ForeignKey("IdAgendamento")]
        public virtual Agendamento? Agendamento { get; set; }
    }
}